python/cudf/cudf/utils/dtypes.py

```diff
@@ -191,6 +191,17 @@ def is_list_dtype(obj):
     )
 
 
+def is_struct_dtype(obj):
```

Comment:
include `pa.StructType` too?

Reply:
I don't think we want that. For example, we don't treat a `pa.DataType(int64)` as a numerical dtype:

```python
In [9]: cudf.utils.dtypes.is_numerical_dtype(pa.type_for_alias("int64"))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-49f433d7f8c9> in <module>
----> 1 cudf.utils.dtypes.is_numerical_dtype(pa.type_for_alias("int64"))

~/cudf/python/cudf/cudf/utils/dtypes.py in is_numerical_dtype(obj)
    113         return False
    114     return (
--> 115         np.issubdtype(obj, np.bool_)
    116         or np.issubdtype(obj, np.floating)
    117         or np.issubdtype(obj, np.signedinteger)

~/rapids-compose/etc/conda/cuda_10.1/envs/rapids/lib/python3.7/site-packages/numpy/core/numerictypes.py in issubdtype(arg1, arg2)
    386     """
    387     if not issubclass_(arg1, generic):
--> 388         arg1 = dtype(arg1).type
    389     if not issubclass_(arg2, generic):
    390         arg2 = dtype(arg2).type

TypeError: Cannot interpret 'DataType(int64)' as a data type

```